tests/test_utils_signal.py

```diff
@@ -77,3 +78,42 @@ def test_error_logged_if_deferred_not_supported(self):
         self.assertEqual(len(l.records), 1)
         self.assertIn("Cannot return deferreds from signal handler", str(l))
         dispatcher.disconnect(test_handler, test_signal)
+
+class SendLogTest(unittest.TestCase):
+    @defer.inlineCallbacks
+    def test_send_deferred(self):
+        test_ok_signal = object()
+        test_error_signal = object()
+        handlers_called = set()
+
+        dispatcher.connect(self.error_handler, signal=test_error_signal)
+        dispatcher.connect(self.ok_handler, signal=test_ok_signal)
+        result = yield defer.maybeDeferred(
+            self._get_result, test_ok_signal, arg='test',
+            handlers_called=handlers_called
+        )
+
+        with pytest.raises(defer.FirstError):
```

Comment:
I tried first with `self.assertRaises` but somehow the `py27:jessies` tests was failing with `TypeError: assertRaises() takes at least 3 arguments (2 given)` which is really weird since python2.7 should support assertRaises with `contextmanager` and is used in this form in other scrapy tests.

Maybe the `yield` in the context does trigger some fixed bugs with `assertRaises`.